src/components/ebay-date-textbox/component.js

```diff
@@ -0,0 +1,136 @@
+// @ts-check
+
+import Expander from 'makeup-expander';
+import { toISO } from '../ebay-calendar/component';
+import * as eventUtils from '../../common/event-utils';
+
+const MIN_WIDTH_FOR_DOUBLE_PANE = 600;
+
+/**
+ * @typedef {import('../ebay-calendar/component').DayISO} DayISO
+ * @typedef {{
+ *   locale?: string,
+ *   range?: boolean,
+ *   disableBefore?: Date | number | string,
+ *   disableAfter?: Date | number | string,
+ *   disableWeekdays?: number[],
+ *   disableList?: (Date | number | string)[],
+ *   getA11yShowMonthText?: (monthName: string) => string,
+ *   a11yRangeStartText?: string,
+ *   a11yInRangeText?: string,
+ *   a11yRangeEndText?: string,
+ * }} Input
+ * @typedef {{
+ *   numMonths: number,
+ *   firstSelected: DayISO | undefined,
+ *   secondSelected: DayISO | undefined,
+ *   popover: boolean,
+ * }} State
+ * @extends {Marko.Component<Input, State>}
+ */
+export default class extends Marko.Component {
+    onCreate() {
+        this.state = {
+            numMonths: 1,
+            firstSelected: undefined,
+            secondSelected: undefined,
+            popover: false,
+        };
+
+        this.calculateNumMonths();
+    }
+
+    onMount() {
+        this.expander = new Expander(/** @type {HTMLElement} */ (this.el), {
+            hostSelector: '.ebay-date-textbox--main > .icon-btn',
+            contentSelector: '.date-textbox__popover',
+            expandOnClick: true,
+            autoCollapse: true,
+        });
+        this.subscribeTo(eventUtils.resizeUtil).on('resize', this.calculateNumMonths.bind(this));
+    }
+
+    onDestroy() {
+        this.expander?.destroy();
+    }
+
+    /**
+     * @param {Input} input
+     */
+    onInput(input) {
+        if (!input.range) {
+            this.state.secondSelected = undefined;
+        }
+    }
+
+    calculateNumMonths() {
+        this.state.numMonths =
+            document.documentElement.clientWidth < MIN_WIDTH_FOR_DOUBLE_PANE ? 1 : 2;
+    }
+
+    /**
+     * @param {number} index
+     * @param {{
+     *   originalEvent: Event,
+     *   value: string,
+     * }} param1
+     */
+    handleInputChange(index, { value }) {
+        const valueDate = new Date(value);
+        const iso = isNaN(valueDate.getTime()) ? undefined : toISO(valueDate);
+        if (index === 0) {
+            this.state.firstSelected = iso;
+        } else {
+            this.state.secondSelected = iso;
+        }
+        this.emitSelectedChange();
+    }
+
+    openPopover() {
+        this.state.popover = true;
+    }
+
+    closePopover() {
+        this.state.popover = false;
+    }
+
+    /**
+     * @param {{ iso: DayISO }} param0
+     */
+    onPopoverSelect({ iso }) {
+        if (this.input.range) {
+            if (this.state.firstSelected && this.state.secondSelected) {
```

Comment:
This can be simplified. There is only 1 else if case where firstSected is not set to iso. Why not set it always and then just set it again in that other state. 
It seems like this can simply be:
Set first state
Check for that one off case and set iso to second state